.github/workflows/check.yml

```diff
@@ -87,24 +88,35 @@ jobs:
             - name: Checkout
               uses: actions/checkout@v2
 
+            - name: Cache native code artifact
+              id: cache-native-code-artifact
+              uses: actions/cache@v2.1.4
+              with:
+                  path: ${{ matrix.config.artifact_path }}
+                  key: ${{ matrix.config.arch }}-${{ runner.os }}-native-helper-${{ matrix.rust-version }}-${{ hashFiles('native-helper/src/**', 'native-helper/Cargo.toml', 'native-helper/Cargo.lock') }}
+
             - name: Set up Rust
+              if: ${{ !steps.cache-native-code-artifact.outputs.cache-hit }}
               uses: actions-rs/toolchain@v1
               with:
                   profile: minimal
-                  toolchain: stable
+                  toolchain: ${{ matrix.rust-version }}
                   target: ${{ matrix.config.target }}
                   default: true
 
-            - name: Cache native code
+            - name: Cache native code dependencies
+              if: ${{ !steps.cache-native-code-artifact.outputs.cache-hit }}
               uses: actions/cache@v2.1.4
               with:
+                  # https://doc.rust-lang.org/cargo/guide/cargo-home.html#caching-the-cargo-home-in-ci
                   path: |
-                      ~/.cargo/registry
-                      ~/.cargo/git
-                      native-helper/target
-                  key: ${{ runner.os }}-native-helper-${{ hashFiles('native-helper/Cargo.lock') }}
+                      ~/.cargo/registry/index
+                      ~/.cargo/registry/cache
+                      ~/.cargo/git/db
+                  key: native-helper-deps-${{ matrix.rust-version }}-${{ hashFiles('native-helper/Cargo.lock') }}
```

Comment:
I released that caching dependencies isn't really important in our case, because we update dependencies more often than update native-helper code. So I just removed this cache